bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1275,27 +1294,30 @@ private void checkStatus() {
                 additionalErrors = List.of("This PR only contains changes already present in the target");
             } else {
                 // Determine current status
-                jcheckType = "jcheck";
-                var additionalConfiguration = AdditionalConfiguration.get(localRepo, localHash,
-                        pr.repository().forge().currentUser(), comments, reviewMerge);
-                checkablePullRequest.executeChecks(localHash, censusInstance, visitor, additionalConfiguration, checkablePullRequest.targetHash());
-                // Don't need to run the second round if confOverride is set.
-                if (workItem.bot.confOverrideRepository().isEmpty() && isFileUpdated(".jcheck/conf", localHash)) {
-                    jcheckType = "second jcheck";
-                    PullRequestCheckIssueVisitor visitor2 = checkablePullRequest.createVisitor(pr.headHash());
-                    log.info("Run jcheck again with the updated configuration");
-                    checkablePullRequest.executeChecks(localHash, censusInstance, visitor2, additionalConfiguration, pr.headHash());
-                    secondJCheckMessage.addAll(visitor2.messages().stream()
+                jcheckType = "target jcheck";
+                checkablePullRequest.executeChecks(localHash, censusInstance, visitor, targetJCheckConf);
+
+                // If the PR updates .jcheck/conf then Need to run JCheck again using the configuration
+                // from the resulting commit. Not needed if we are overriding the JCheck configuration since
+                // then we won't use the one in the repo anyway.
+                if (workItem.bot.confOverrideRepository().isEmpty() &&
+                    (isFileUpdated(Path.of(".jcheck").resolve("conf"), localHash) || isJCheckConfUpdatedInMergePR)) {
```

Comment:
Style nit. I think this way of creating a path looks a bit overly complex. I would suggest either just `Path.of(".jcheck/conf")` or split the elements like this `Path.of(".jcheck", "conf")`.

Reply:
Sure, fixed in latest